roles/generate-jenkins/templates/Jenkinsfile.j2

```diff
@@ -521,6 +521,35 @@ pipeline {
               else
                 echo "No templates to delete"
               fi
+{% if full_custom_readme is not defined and (init_diagram is not defined or init_diagram) %}
+              echo "Starting Stage 2.5 - Update init diagram"
+              if ! grep -q 'init_diagram:' readme-vars.yml; then
+                echo "Adding the key 'init_diagram' to readme-vars.yml"
+                sed -i '\\|^#.*changelog.*$|d' readme-vars.yml 
+                sed -i 's|^changelogs:|# init diagram\\ninit_diagram:\\n\\n# changelog\\nchangelogs:|' readme-vars.yml
+              fi
+              docker run -d --rm -v ${TEMPDIR}/d2:/output -e PUID=$(id -u) -e PGID=$(id -g) ghcr.io/linuxserver/d2-builder:latest ${CONTAINER_NAME}:{{ release_tag }}
+              yq -ei ".init_diagram |= load_str(\\"${TEMPDIR}/d2/${CONTAINER_NAME}-{{ ls_branch }}.d2\\")" readme-vars.yml || :
+              if [[ $(grep -hs ^ readme-vars.yml | md5sum | cut -c1-8) != $(grep -hs ^ ${TEMPDIR}/docker-${CONTAINER_NAME}/readme-vars.yml | md5sum | cut -c1-8) ]]; then
```

Comment:
Is there are reason for the grep rather than just doing something like `md5sum readme-vars.yml | cut -c1-8`

Reply:
Copy pasta from the other check where we check the cumulative hash of a list of files

Reply:
simplified